torch/_refs/nn/functional/__init__.py

```diff
@@ -294,3 +295,37 @@ def tanhshrink(a: TensorLikeType) -> TensorLikeType:
             "Expected a tensor input for an elementwise unary operation!"
         )
     return refs.sub(a, refs.tanh(a))
+
+
+@elementwise_unary_scalar_wrapper
+@elementwise_type_promotion_wrapper(
+    type_promoting_args=("a"),
+    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT,
+)
+def hardtanh(
+    a: TensorLikeType,
+    min_val: NumberType = -1.0,
```

Comment:
haha, well if these params are typed as numbers then the values can be -1 and 1 again ;)

Reply:
https://pytorch.org/docs/stable/generated/torch.nn.functional.hardtanh.html

I thought it made sense to be 1.0/-1.0, since that's what the documentation shows.

Reply:
It's more future-proof to have them be -1 and 1 if this function intends to support integer tensor inputs